src/main/java/org/threeten/extra/chrono/InternationalFixedDate.java

```diff
@@ -0,0 +1,887 @@
+     * <p/>
+     * This returns the length of the year in days.
+     * Year lengths match those of the ISO calendar system.
+     *
+     * @return the length of the year in days: 365 or 366
+     */
+    @Override
+    public int lengthOfYear() {
+        return InternationalFixedChronology.DAYS_IN_YEAR + (isLeapYear() ? 1 : 0);
+    }
+
+    @Override
+    public InternationalFixedDate plus(final long amountToAdd, final TemporalUnit unit) {
+        if (amountToAdd == 0) {
+            return this;
+        }
+
+        if (unit instanceof ChronoUnit) {
+            ChronoUnit f = (ChronoUnit) unit;
+
+            switch (f) {
+                case DAYS:
+                    return plusDays(amountToAdd);
+                case MONTHS:
+                    return plusMonths(amountToAdd);
+                case YEARS:
+                    return plusYears(amountToAdd);
+                default:
+                    break;
+            }
+        }
+
+        return (InternationalFixedDate) super.plus(amountToAdd, unit);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    InternationalFixedDate plusYears(final long yearsToAdd) {
+        if (yearsToAdd == 0) {
+            return this;
+        }
+
+        int newYear = (int) Math.addExact(getProlepticYear(), yearsToAdd);
+
+        if (isYearDay()) {
+            return create(newYear, false, true);
+        }
+
+        int day = getDayOfMonth();
+        int newMonth = getCalculatedMonth();
+
+        if (isLeapDay()) {
+            if ((yearsToAdd & 3) == 0) {
+                return create(newYear, true, false);
+            }
+
+            day = 1;
+        }
+
+        return create(newYear, newMonth, day);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public InternationalFixedDate plusMonths(final long months) {
+        if (months == 0) {
+            return this;
+        }
+
+        int day = isYearDay() ? InternationalFixedChronology.DAYS_IN_MONTH : getCalculatedDayOfMonth();
+        int newMonth = (int) Math.addExact(getCalculatedProlepticMonth(), months);
+        int year = newMonth / InternationalFixedChronology.MONTHS_IN_YEAR;
+        newMonth = 1 + (newMonth % InternationalFixedChronology.MONTHS_IN_YEAR);
+
+        return create(year, newMonth, day);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public InternationalFixedDate plusDays(final long daysToAdd) {
+        if (daysToAdd == 0) {
+            return this;
+        }
+
+        return ofEpochDay(toEpochDay() + daysToAdd);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public ValueRange range(final TemporalField field) {
+        boolean special = isYearDay() || isLeapDay();
+
+        if (field instanceof ChronoField) {
+            if (isSupported(field)) {
+                ChronoField f = (ChronoField) field;
+
+                switch (f) {
+                    case ALIGNED_DAY_OF_WEEK_IN_MONTH:
```

Comment:
For the aligned day/week ranges, IFC also returns the empty range (0,0) for leap-day or year-day.
Same for day-of-week.
What exactly is not correct ?